Adding Glossary

AlannaBurke opened the pull request:
Fresh PR for glossary feature. 

cc @svekars @sekyondaMeta

svekars commented on docs/source/conf.py:
```diff
@@ -74,6 +75,18 @@
+
+# Skip all URLs except glossary term links (glossary.html#term-*)
+tippy_skip_urls = (r"^(?!.*glossary\.html#term-).*$",)
+tippy_enable_mathjax = True
```
```suggestion
```

svekars commented on docs/source/jit.rst:
```diff
@@ -1,5 +1,5 @@
 TorchScript
-===========
```
Undo all changes to this file?

svekars commented on docs/source/glossary.md:
```diff
@@ -0,0 +1,59 @@
+(glossary)=
+# PyTorch Glossary
+
+This glossary provides definitions for terms commonly used in PyTorch documentation.
```
@albanD - can you help to review the definitions?

AlannaBurke replied:
FYI this file with the definitions was already there, I haven't added any new ones yet, I don't think.

albanD replied:
The glossary.md is 5 years old so very much not accurate. Left comments below to update.

albanD commented on docs/source/glossary.md:
```diff
@@ -0,0 +1,59 @@
+
+Kernel
+   Implementation of a PyTorch operation, specifying what should be done when an
+   operation executes.
```
Refers to the "on device" implementation of a given operation. Reffering to the part of the Operation that actually computes the result.

albanD commented on docs/source/glossary.md:
```diff
@@ -0,0 +1,59 @@
+Leaf Operation
+   An operation that's considered a basic operation, as opposed to a {term}`Compound
+   Operation`. Leaf Operation always has dispatch functions defined, usually has a
+   derivative function defined as well.
```
I'm not sure what this refers to. I would remove

albanD commented on docs/source/glossary.md:
```diff
@@ -0,0 +1,59 @@
+   Operation`. Leaf Operation always has dispatch functions defined, usually has a
+   derivative function defined as well.
+
+Device Kernel
```
I would only use the Kernel above

svekars commented on .ci/docker/requirements-docs.txt:
```diff
@@ -2,7 +2,8 @@ sphinx==7.2.6
 
-pytorch_sphinx_theme2==0.2.0
+git+https://github.com/pytorch/pytorch_sphinx_theme.git@5a164c28a88e2b0b9cd483be800901a95d1442225#egg=pytorch_sphinx_theme2
+#pytorch_sphinx_theme2==0.2.0
```
```suggestion
git+https://github.com/pytorch/pytorch_sphinx_theme.git@a164c28a88e2b0b9cd483be800901a95d1442225#egg=pytorch_sphinx_theme2
#pytorch_sphinx_theme2==0.2.0
```

svekars commented:
@pytorchbot rebase

svekars commented on docs/source/user_guide/torch_compiler/export.md:
```diff
@@ -56,7 +56,7 @@ Under the hood, `torch.export` leverages the following latest technologies:
   provides a massively improved graph capturing experience, with much fewer
   rewrites needed in order to fully trace the PyTorch code.
-- **AOT Autograd** ensures the graph is decomposed/lowered to the ATen operator
+- **AOT Autograd** ensures the graph is decomposed/lowered to the {term}`ATen` operator
```
```suggestion
- **AOT Autograd** ensures the graph is decomposed or lowered to the {term}`ATen` operator
```

AlannaBurke committed:
Add interactive glossary with hover tooltips

- Add glossary.md and _glossary.md with PyTorch terminology definitions
- Configure sphinx_tippy for hover tooltips on glossary terms
- Add glossary term references ({term}) across docs (export, compiler, onnx)
- Remove old GLOSSARY.md in favor of Sphinx-integrated glossary
- Update requirements-docs.txt with sphinx_tippy dependency
- Update index.md to include glossary in toctree

AlannaBurke committed:
Restoring accidentally removed lines.

AlannaBurke committed:
Merge branch 'main' into glossary-fresh

AlannaBurke commented on docs/source/_glossary.md:
```diff
@@ -0,0 +1,59 @@
+
+Custom Operation
+   An Operation that is defined by users and is usually a {term}`Compound Operation`.
+   For example, this [tutorial](https://pytorch.org/docs/stable/notes/extending.html)
```
```suggestion
   For example, this [tutorial](https://docs.pytorch.org/tutorials/advanced/custom_ops_landing_page.html)
```

AlannaBurke commented on docs/source/_glossary.md:
```diff
@@ -0,0 +1,59 @@
+   details how to create Custom Operations.
+
+Kernel
+   Implementation of a PyTorch operation, specifying what should be done when an
```
```suggestion
    Refers to the "on device" implementation of a given operation. Reffering to the part of the Operation that actually computes the result.
```

AlannaBurke commented on docs/source/_glossary.md:
```diff
@@ -0,0 +1,59 @@
+
+Kernel
+   Implementation of a PyTorch operation, specifying what should be done when an
+   operation executes.
```
```suggestion

```

AlannaBurke commented on docs/source/_glossary.md:
```diff
@@ -0,0 +1,59 @@
+   Implementation of a PyTorch operation, specifying what should be done when an
+   operation executes.
+
+Compound Operation
```
```suggestion
Composite Operation
```

AlannaBurke commented on docs/source/_glossary.md:
```diff
@@ -0,0 +1,59 @@
+Non-Leaf Operation
+   Same as {term}`Compound Operation`.
+
+Leaf Operation
```
```suggestion
```

AlannaBurke commented on docs/source/_glossary.md:
```diff
@@ -0,0 +1,59 @@
+   Same as {term}`Compound Operation`.
+
+Leaf Operation
+   An operation that's considered a basic operation, as opposed to a {term}`Compound
```
```suggestion
```

AlannaBurke commented on docs/source/_glossary.md:
```diff
@@ -0,0 +1,59 @@
+
+Leaf Operation
+   An operation that's considered a basic operation, as opposed to a {term}`Compound
+   Operation`. Leaf Operation always has dispatch functions defined, usually has a
```
```suggestion
```

AlannaBurke commented on docs/source/_glossary.md:
```diff
@@ -0,0 +1,59 @@
+Leaf Operation
+   An operation that's considered a basic operation, as opposed to a {term}`Compound
+   Operation`. Leaf Operation always has dispatch functions defined, usually has a
+   derivative function defined as well.
```
```suggestion
```

AlannaBurke commented on docs/source/_glossary.md:
```diff
@@ -0,0 +1,59 @@
+Device Kernel
+   Device-specific kernel of a {term}`Leaf Operation`.
+
+Compound Kernel
```
```suggestion
```

AlannaBurke commented on docs/source/_glossary.md:
```diff
@@ -0,0 +1,59 @@
+   Device-specific kernel of a {term}`Leaf Operation`.
+
+Compound Kernel
+   Opposed to {term}`Device Kernels<Device Kernel>`, Compound kernels are usually
```
```suggestion
```

AlannaBurke commented on docs/source/_glossary.md:
```diff
@@ -0,0 +1,59 @@
+
+Compound Kernel
+   Opposed to {term}`Device Kernels<Device Kernel>`, Compound kernels are usually
+   device-agnostic and belong to {term}`Compound Operations<Compound Operation>`.
```
```suggestion
```

AlannaBurke commented on docs/source/_glossary.md:
```diff
@@ -0,0 +1,59 @@
+   Just-In-Time Compilation.
+
+Tracing
+   Using `torch.jit.trace` on a function to get an executable that can be optimized
```
```suggestion
In PyTorch, tracing is a way to convert a PyTorch model (or function) into a static computation graph by running it once (or a few times) with example inputs and recording (“tracing”) the tensor operations that actually execute.```

AlannaBurke commented on docs/source/_glossary.md:
```diff
@@ -0,0 +1,59 @@
+
+Tracing
+   Using `torch.jit.trace` on a function to get an executable that can be optimized
+   using just-in-time compilation.
```
```suggestion
```

AlannaBurke committed:
Fixing menu.

AlannaBurke committed:
Merge branch 'glossary-fresh' of github.com:AlannaBurke/pytorch into glossary-fresh

AlannaBurke committed:
Update docs/source/_glossary.md

AlannaBurke committed:
Update docs/source/_glossary.md

AlannaBurke committed:
Update docs/source/_glossary.md

AlannaBurke committed:
Update docs/source/_glossary.md

AlannaBurke committed:
Update docs/source/_glossary.md

AlannaBurke committed:
Update docs/source/_glossary.md

AlannaBurke committed:
Update docs/source/_glossary.md

AlannaBurke committed:
Update docs/source/_glossary.md

AlannaBurke committed:
Update docs/source/_glossary.md

AlannaBurke commented on docs/source/_glossary.md:
```diff
@@ -0,0 +1,53 @@
+    Refers to the "on device" implementation of a given operation. Reffering to the part of the Operation that actually computes the result.
+
+Composite Operation
+   A Compound Operation is composed of other operations. Its kernel is usually
```
```suggestion
   A Composite Operation is composed of other operations. Its kernel is usually
```

AlannaBurke committed:
Updating glossary.

AlannaBurke committed:
Updating Glossary.

AlannaBurke committed:
Merge branch 'glossary-fresh' of github.com:AlannaBurke/pytorch into glossary-fresh
